### PR TITLE
feat: implement configure() pipeline function

### PR DESCRIPTION
**Pull request by S-Leuthold**
## Summary

- Implements `configure()` — the bridge between data preparation and model evaluation
- Builds a configuration grid (Cartesian product of models × transformations × preprocessing × feature selection × covariate sets)
- Promotes one response variable to `role = "outcome"` for modeling
- Supports covariate power-set expansion for benchmarking across covariate subsets
- Reconfiguration-safe: calling twice overwrites cleanly, reverting previous outcome

## Changes

- **R/pipeline-configure.R** — new file, ~330 lines. Main function + two internal helpers (`generate_config_id`, `generate_power_set`)
- **R/constants.R** — short model names (`rf` not `random_forest`), `MODEL_SPECS` lookup table, added `log10` to valid transformations
- **R/class-core.R** — `print()` shows outcome variable, `summary()` shows expansion axes when configured, next-step logic includes `validate()` between configure and evaluate
- **DESCRIPTION** — added `pipeline-configure.R` to Collate
- **tests/testthat/test-pipeline-configure.R** — 89 tests across 9 categories (validation, outcome promotion, config grid, covariate expansion, storage, fusion, reconfiguration, CLI output, edge cases)

## Test plan

- [x] 89 new tests passing (0 failures)
- [x] 98% coverage on `pipeline-configure.R`
- [x] Full suite: 450 tests passing, 0 failures
- [ ] CodeRabbit review

## Summary by CodeRabbit

* **New Features**
  - Pipeline configuration tool to build, validate and store model/processing grids; supports model selection, transformations, preprocessing, feature selection, covariate expansion, and CV/tuning settings.
  - Added compact model names and additional model options (including MLP and boosted variants); log10 transformation available.

* **Improvements**
  - Summaries and printouts now show explicit Outcome details, expansion metadata, and updated next-step guidance including validation.

* **Tests**
  - Extensive tests for configuration, covariate expansion, outcome handling, CLI output, and reconfiguration behavior.

* **Chores**
  - Packaging metadata order updated for inclusion of configuration tooling.

### CODE REVIEW

File: R/class-core.R [hunk 1]
```diff
@@ -511,6 +511,18 @@ print.horizons_data <- function(x, ...) {
...
+    has_outcome <- any(x$data$role_map$role == "outcome")
+
+    if (has_outcome) {
+
+      outcome_var <- x$data$role_map$variable[x$data$role_map$role == "outcome"]
+      has_more    <- has_responses || has_covars
+      branch      <- if (has_more) "\u251C\u2500" else "\u2514\u2500"
+      cat(paste0("   ", branch, " Outcome: ", outcome_var, "\n"))
+
+    }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/class-core.R:523]: ✅ Fixed — moved `has_outcome` computation before the predictors block and included it in `has_more` so the tree branches correctly when outcome is the only optional section.

File: R/pipeline-configure.R [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+  }
+
+  if (!is.numeric(bayesian_iter) || bayesian_iter < 0) {
+
+    abort_nested(
+      "`bayesian_iter` must be non-negative",
+      c(paste0("Got: ", bayesian_iter))
+    )
+
+  }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/pipeline-configure.R]: ✅ Fixed — added `length() != 1` and `is.na()` checks to all three tuning parameter validations. Also using `deparse()` in error messages for better vector/NA display.

File: R/pipeline-configure.R [hunk 3]
```diff
@@ -0,0 +1,607 @@
...
+  if (!is.null(x$config$configs)) {
+
+    warning("Overwriting previous configuration", call. = FALSE)
+
+    x$data$role_map$role[x$data$role_map$role == "outcome"] <- "response"
+
+    ## Recalculate after reset
+    responses <- x$data$role_map$variable[x$data$role_map$role == "response"]
+
+  }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/pipeline-configure.R]: ✅ Fixed — reconfiguration now clears validation, evaluation, models, and ensemble state to prevent stale downstream artifacts.

File: R/pipeline-configure.R [hunk 4]
```diff
@@ -0,0 +1,607 @@
...
+  config_grid <- config_grid[, c(
+    "config_id", "model", "transformation", "preprocessing",
+    "feature_selection", "covariates",
+    "preprocessing_params", "feature_params", "transform_params"
+  )]
+
+  ## Deduplicate (defensive) ---------------------------------------------------
+
+  config_grid <- config_grid[!duplicated(config_grid$config_id), ]
+
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/pipeline-configure.R:478]: Dismissing — the math here is incorrect. `tidyr::crossing()` already deduplicates its inputs, so the grid cannot contain duplicate rows. The hash only needs to distinguish rows within a single grid, not globally. A 6-hex hash (16M combinations) is more than sufficient for grids that max out at a few thousand configs. The dedup on `config_id` is a defensive safety net, not the primary uniqueness mechanism.
